validation.md

```diff
@@ -0,0 +1,152 @@
+# Improving Validation
+
+The goal of this change is to improve the validation in Promitor to make it easier
+for users to find problems with their configuration, and to make it easier to add
+validation into the deserializers.
+
+## Must Haves
+
+- A consistent output format that explains exactly what the problem is, and ideally
+  where it is.
+- We must indicate when unknown field names are specified so we can highlight typos.
+
+## Nice to Haves
+
+- A schema for the configuration files so we have a single source of truth for the
+  configuration format.
+- Contextual output showing snippets of the parts of the file that have issues.
+- Suggestions for typos (e.g. 'tneantId' instead of 'tenantId').
+
+## Output Format
+
+To make it easy for users to fix problems with the configuration file, we need to
+tell them the filename, line number and ideally the column where the error occurs.
+
+To achieve this we will use the following format:
+
+```text
+<filename>:<line-number>:<column>: <message>
+```
+
+We may want to remove the filename from the error messages or make it optional since
```

Comment:
I like this approach more indeed!

Reply:
Yeah - I agree. I just figured I'd leave both in this doc temporarily to brainstorm a bit.